client/views/voip/VoIPLayout.tsx

```diff
@@ -164,44 +239,26 @@ class VoIPLayout
 		if (element) {
 			element.style.display = 'none';
 		}
-
-		if (this.userName.current?.textContent) {
-			this.config.authUserName = this.userName.current?.textContent;
-		} else {
-			this.config.authUserName = this.userName.current?.defaultValue;
-		}
-		if (this.password.current?.textContent) {
-			this.config.authPassword = this.password.current?.textContent;
-		} else {
-			this.config.authPassword = this.password.current?.defaultValue;
-		}
-		if (this.registrar.current?.textContent) {
-			this.config.sipRegistrarHostnameOrIP = this.registrar.current?.textContent;
-		} else {
-			this.config.sipRegistrarHostnameOrIP = this.registrar.current?.defaultValue;
-		}
-		if (this.webSocketPath.current?.textContent) {
-			this.config.webSocketURI = this.webSocketPath.current?.textContent;
-		} else {
-			this.config.webSocketURI = this.webSocketPath.current?.defaultValue;
+		if (!this.userHandler) {
+			try {
+				await this.initUserAgent();
+				this.setState({ isReady: true });
+			} catch (error) {
+				this.logger?.error('componentDidMount() Error in getting extension Info', error);
```

Comment:
`this.logger` isn't optional. Can we remove the `?` ?

Reply:
Fixed